modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -702,14 +748,31 @@ public boolean setTabSize(int spaces) {
      *                                                                         *
      **************************************************************************/
 
-    private int getLineIndex(float y) {
+    private int getLineIndex(float y, String text) {
         int index = 0;
         float bottom = 0;
+        boolean isTextPresent = text == null ? true : false;
```

Comment:
this looks very confusing to me... what text is present?

Reply:
I agree that the naming and the conditionals were confusing. So optimized the code for better readability.